### ISSUE TITLE
gem push fails with Broken Pipe IOError

### ISSUE THREAD

**Issue by AndyObtiva** [jruby/jruby#6871]
**Environment Information**
  
Provide at least:
* JRuby version (`jruby -v`) and command line (flags, JRUBY_OPTS, etc): `9.3.0.0`
* Operating system and platform (e.g. `uname -a`): `Darwin Andys-MacBook-Pro.local 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:07:06 PST 2021; root:xnu-7195.81.3~1/RELEASE_X86_64 x86_64`

Other relevant info you may wish to add:
* `gem --version` returns 3.2.28

**Expected Behavior**

Running `gem push path_to_gem.gem` should result in releasing the gem to RubyGems.org successfully.

This works in `jruby-9.2.19.0`

**Actual Behavior**

In `jruby-9.3.0.0`, I get a broken pipe IOError when running `gem push path_to_gem.gem`:

```
% gem push pkg/glimmer-dsl-swt-4.21.0.1.gem 
Pushing gem to https://rubygems.org...
ERROR:  While executing gem ... (IOError)
    Broken pipe
```

It works fine in `ruby-3.0.2` and `jruby-9.2.19.0`. My Internet works well and the same in all cases.

**Comment by headius** [comment]
Could you try to run this with more debugging output, via some combination of `--verbose` and `--backtrace` and `--debug`?

I'm trying to figure out if there's a dummy RG server somewhere I could use to test this. If you can get more debugging output we might have enough information to narrow the scope of this issue.

**Comment by headius** [comment]
Oh and thank you for the extra logging output! Seems like this ties the issue back to jruby-openssl. However, I think we shipped the same jruby-openssl in 9.2.19.0 so there's more to the story.

I was able to push a gem with the 9.3.0.0 release (installed via rvm) so we need to dig deeper on why this is failing for you.

```
[] ~/projects/weakling $ jruby -v
jruby 9.3.0.0 (2.6.8) 2021-09-17 85c20e780f OpenJDK 64-Bit Server VM 25.222-b10 on 1.8.0_222-b10 +jit [darwin-x86_64]

[] ~/projects/weakling $ gem -v
3.2.14
```

@kares Any thoughts here?

**Comment by AndyObtiva** [comment]
Yeah, sorry I forgot to provide the full version line, which indicates I am using JDK16.

```
% jruby -v
jruby 9.3.0.0 (2.6.8) 2021-09-17 85c20e780f Java HotSpot(TM) 64-Bit Server VM 16.0.2+7-67 on 16.0.2+7-67 +jit [darwin-x86_64]
```

```
% gem -v
3.2.28
```

**Comment by headius** [comment]
I will give that a shot. I also have a macOS 11 system here that I'll try to fire up tomorrow.

**Comment by michaelklishin** [comment]
I ran into this with JDK 13.0.2 on MacOS 11.6 (Intel hardware).

**Comment by AndyObtiva** [comment]
Gem pushing with JRuby-9.3.1.0 on Linux also works. Given that it works on Windows too, that leaves only the Mac (x64) as a culprit.
